spec/javascripts/actions/story_spec.js

```diff
@@ -219,4 +223,110 @@ describe('Story Actions', () => {
       expect(fakeDispatch).toHaveBeenCalledWith(Story.updateHighlight(storyId, true));
     });
   });
+
+  describe('confirmBeforeSaveIfNeeded', () => {
+    let story;
+
+    beforeEach(() => {
+      story = storyFactory();
+    });
+
+    describe('when do not need of confirmation', () => {
+      const needConfirmation = sinon.stub().returns(false);
+      const confirm = sinon.stub();
+
+      describe('and callback.onConfirmed do not throws an error', () => {
+        it('call callback.onConfirmed', async () => {
+          const callback = { onConfirmed: sinon.stub() };
+
+          await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
+          expect(callback.onConfirmed).toHaveBeenCalled();
+        });
+      });
+
+      describe('and callback.onConfirmed throws an error', () => {
+        let error;
+        let callback;
+
+        beforeEach(async () => {
+          error = { error: "boom" };
+          callback = { 
+            onConfirmed: sinon.stub().rejects(error), 
+            onError: sinon.stub() 
+          };
+        });
+
+        it('call callback.onConfirmed', () => {
```

Comment:
It won't work if we don't do that.

```suggestion
        it('call callback.onConfirmed', async () => {
```

Reply:
Oh, sure I forgot that.